### PR TITLE
color functions calls; update test

### PR DESCRIPTION
**Pull request by hgGeorg**
Since #56 hasn't received any updates for a while, I decided to create an updated version.
It basically is the same pull request but doesn't break keywords like `class`.
Test is also updated to expect the correct token.

### CODE REVIEW

File: spec/coffee-script-spec.coffee [hunk 1]
```diff
@@ -247,3 +247,18 @@ describe "CoffeeScript grammar", ->
     expect(lines[0][0]).toEqual value: '`', scopes: ["source.coffee", "string.quoted.script.coffee", "punctuation.definition.string.begin.coffee"]
     expect(lines[0][1]).toEqual value: 'v', scopes: ["source.coffee", "string.quoted.script.coffee", "constant.character.escape.coffee"]
     expect(lines[1][0]).toEqual value: 'a ', scopes: ["source.coffee", "variable.assignment.coffee", "variable.assignment.coffee"]
+
+  it "tokenizes functions", ->
+    {tokens} = grammar.tokenizeLine("foo = -> 1")
+    expect(tokens[0]).toEqual value: "foo ", scopes: ["source.coffee", "meta.function.coffee", "entity.name.function.coffee"]
+
+    {tokens} = grammar.tokenizeLine("foo bar")
+    expect(tokens[0]).toEqual value: "foo ", scopes: ["source.coffee", "entity.name.function.coffee"]
```
Comment (50Wliu) [inline spec/coffee-script-spec.coffee]: Why is the space here also tokenized as a function?

Comment (hgGeorg) [inline spec/coffee-script-spec.coffee]: `foo bar` is a function call where `foo` is the function and `bar` the argument.

Comment (hgGeorg) [inline spec/coffee-script-spec.coffee]: Hmm... after checking existing tests, I've noticed that in the `a = b` test case, the space behind `a` is also tokenized as variable assignment.

That's why it didn't seem wrong to me in the first place.

Please let me know if my assumption is/was correct.

Comment (50Wliu) [inline spec/coffee-script-spec.coffee]: Technically, that's also wrong I think.  Really only the actual variable/function/whatever should be tokenized.

If I merge this as is (for consistency), can you make a new PR that fixes those whitespace problems?

Comment (hgGeorg) [inline spec/coffee-script-spec.coffee]: I can't promise that but I'll do if time constraints allow it.

**Comment by 50Wliu** [comment]
@hgGeorg Any other changes you want to make or do you think this is ready?

**Comment by 50Wliu** [comment]
Actually, I just tested this out, and `hello world` seems to match.  That doesn't seem like a function call to me :laughing:.

**Comment by MaximSokolov** [comment]
> Actually, I just tested this out, and hello world seems to match. That doesn't seem like a function call to me :laughing:.

`hello` is function name, `world` is argument, no?

**Comment by 50Wliu** [comment]
> hello is function name, world is argument, no?

Ugh, I'm in JavaScript world again, ignore me.

**Comment by hgGeorg** [comment]
@50Wliu Nothing I'm currently aware of.

**Pull request merged**